utils/cache.js

```diff
@@ -0,0 +1,113 @@
+const CACHEEXPIRYTIME = 2; // Default cache expriy time in Minutes
+const CACHESIZE = 10; // Default maximum size for cache pool (in Megabytes)
+const minutesToMilliseconds = (minutes) => minutes * 60000; // Converts given minutes into milliseconds
+
+/**
+ * Cache pool to get and store API responses
+ * @param {Object} [opt] options for cache pool
+ * @param {number} opt.maximumSize Maximum size of the cache pool in Megabytes (MB)
+ */
+const cachePool = (opt = { maximumSize: CACHESIZE }) => {
+  const cacheStore = new Map();
+  let hits = 0;
+
+  /**
+   * Get an API Respnose from cacheStore.
+   * @param {string} key
+   * @returns {null | object}
+   */
+  const get = (key) => {
+    const apiData = cacheStore.get(key);
+
+    if (!apiData) {
+      return null;
+    }
+
+    const isApiDataExpired = new Date().getTime() > apiData.expiry;
+
+    // If data is expired remove it from store, time to get a fresh copy.
+    if (isApiDataExpired) {
+      cacheStore.delete(key);
```

Comment:
Why are we creating this cache model ourselves rather than using any npm lib for this?

Reply:
We can use something like: https://www.npmjs.com/package/node-cache...

Reply:
I had a look at this package and was thinking to use it at the start, but I think we will have to write a wrapper around it, later on, to implement features like: 
1. Priority flags
2. custom clean-up function logic which depends on priority flags 
3. ability to set maximum memory size for cache pool (node-cache has something similar with maxKeys which allows us to have a limit on the number of keys that can be stored but not the amount of memory cache pool takes)

And hence thought to go ahead with our own cache module.

Also, do we really need a package like `node-cache` when we will probably be not making use of the majority of its features at the current stage?

You can refer to issue ticket #698  which would give a clear understanding of the features I want to build for V1 of this module.

Reply:
👍